app/js/lib/foxbox/foxbox.js

```diff
@@ -123,49 +127,66 @@ export default class Foxbox extends Service {
    * @returns {Promise}
    * @private
    */
-  _initDiscovery() {
+  _initDiscovery(timeout) {
```

Comment:
__note:__ not required for this PR, but just wanted to note that I'd rather simplify this method (remove this nasty outer ```new Promise``` and no need in ```timeout``` argument) and reduce the scope for settimeout:
```
_initDiscovery() {
    // For development purposes if you want to skip the
    // discovery phase set the 'foxbox-skipDiscovery' variable to
    // 'true'.
    if (this[p.settings].skipDiscovery) {
      return Promise.resolve();
    }

    return this[p.net].fetchJSON(this[p.settings].registrationService)
      .then(boxes => {
        boxes = Array.isArray(boxes) ? boxes : [];

        ......

        // If the registration server didn't give us any info and
        // we have no record of previous registrations, we schedule
        // a retry.
        if (!this[p.boxes].length &&
          !this[p.settings]._localOrigin &&
          !this[p.settings]._remoteOrigin) {
          console.warn('No boxes found. Retrying...');
          
          return new Promise((resolve) => {
            setTimeout(() => {
              this._initDiscovery().then(resolve, resolve);
            }, 1000);
          });
        }

        ....

        this._dispatchEvent('box-online', true);
      })
      .catch(() => {
        ....
      });
  }
```